luda-editor/luda-editor-client/src/app/editor/timeline/timeline_body/track_body/subtitle_track_body/subtitle_clip_body.rs

```diff
@@ -153,12 +130,251 @@ impl SubtitleClipBody {
             x,
             0.0,
             render![
-                path(stroke_path.clone(), stroke_border_paint),
-                head_rendering_tree,
+                namui::path(clip_body_path.clone(), fill_paint),
+                namui::path(clip_body_path.clone(), stroke_paint)
+                    .with_mouse_cursor(MouseCursor::Grab)
+                    .attach_event(|builder| {
+                        let clip_id = props.clip.id.clone();
+                        builder.on_mouse_down(move |event| {
+                            namui::event::send(EditorEvent::SubtitleClipHeadMouseDownEvent {
+                                clip_id: clip_id.clone(),
+                                click_in_time: timeline_start_at
+                                    + PixelSize(event.local_xy.x + x) * time_per_pixel,
+                            });
+                        })
+                    }),
+                render_text_box(
+                    &subtitle_text,
+                    &head_circle_center_xy,
+                    &tail_circle_center_xy,
+                    circle_radius
+                ),
                 check_flag,
-                tail_rendering_tree,
-                path(stroke_path, stroke_fill_paint),
             ],
         )
     }
 }
+
+struct SubtitleClipBodyRenderSetting {
+    clip_wh: Wh<f32>,
+    circle_radius: f32,
+    head_left_top: Xy<f32>,
+    tail_right_bottom: Xy<f32>,
+    head_height: f32,
+    head_circle_center_xy: Xy<f32>,
+    tail_height: f32,
+    tail_circle_center_xy: Xy<f32>,
+}
+
+fn render_debug_xy(xy: &Xy<f32>, color: Color) -> RenderingTree {
+    path(
+        namui::PathBuilder::new().add_rect(&LtrbRect {
+            left: xy.x,
+            top: xy.y,
+            right: xy.x + 1.0,
+            bottom: xy.y + 1.0,
+        }),
+        namui::PaintBuilder::new()
+            .set_color(color)
+            .set_style(PaintStyle::Fill),
+    )
+}
+
```

Comment:
```suggestion
```
This function is unused. Is it a function that should be reserved for debug? if not, we can remove this